csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +69,38 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public void RemoveField(int fieldIndex)
+        {
+            if (fieldIndex < 0 || fieldIndex > _fields.Count)
```

Comment:
Should be `fieldIndex >= _fields.Count`, right? If it `== _fields.Count`, it would throw below.